### ISSUE TITLE
Broken markup with multi-inline markup check list and highlight

### ISSUE THREAD

**Issue by ubitux** [vimwiki/vimwiki#971]
I was playing with the recently introduced fixes in https://github.com/vimwiki/vimwiki/issues/709 and came across this broken scenario:

With markdown markup, `let g:vimwiki_hl_cb_checked = 1`, and the following content:

```
- [ ] Lorem ipsum dolor sit amet, consectetur adipiscing elit, `sed do eiusmod
      tempor incididunt` ut labore et dolore magna aliqua
- [ ] Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi
      ut aliquip ex ea commodo consequat
```

Checking the first list entry will break the markup for everything that follows.

It was recently pointed out to me by @tinmarino that `let g:vimwiki_hl_cb_checked = 2` was what I was looking for proper multi line support, and it indeed fixes/workaround the issue. But the recent support for multiple inline markup made the `1` mode even more awful than it was previously, so my question is: shouldn't we drop the `1` mode in favor of `2` by default, always? I don't see in what scenario the mode `1` would make any sense anyway.

If it's decided that `1` mode makes sense, then maybe the markup breakage regression should be addressed.
```
Os: Linux
Vim: 800
Branch: dev
Revision: fc056cf
Date: 2020-08-04 01:17:34 -0400
```

**Comment by tinmarino** [comment]
Salut Ubitux,

Very nice NitPicking ! You spot a well hidden bug **Thanks** for that.
As I included the changes, **I will fix that** (and include test from your issue).

> shouldn't we drop the 1 mode in favor of 2 by default, always? 

The default is 0, so I think, it is cool to have "more than boolean" configurations variables, meaning. **I vote to keep** 0,1,2 and to document better.

@+
Tinmarino.

**Comment by tinmarino** [comment]
Fixed: 825870ec1b845

Adding the nested regions in `contains` field of the DoneTask syntax region definition. SO the contained is highliht as before checking DONE.

Note that this is not perfect: we could embed a VimwikiDoneBoldItalicUnderlineStrikethought and a VimwikiDoneItalic and ... but this would be a huge complexity `len([bold, italic, underline, done, strikethrough) ! = 5! = 120` at each startup for a not so used feature.

But at least, this does not break the rest of the file

* `let g:vimwiki_hl_cb_checked = 1`

![Screenshot from 2020-08-04 17-54-35_wiki](https://user-images.githubusercontent.com/6123962/89349063-95287700-d67b-11ea-8331-d59cb9c6057c.png)

* `let g:vimwiki_hl_cb_checked = 2` (No hightlight, no coceal, no bug)

![Screenshot from 2020-08-04 17-55-13-wiki2](https://user-images.githubusercontent.com/6123962/89349156-bd17da80-d67b-11ea-9666-850aefcfb0de.png)

* raw text

```text
Normal syntax
- [X] Lorem ipsum dolor __sit__ amet, consectetur adipiscing elit, `sed do eiusmod
      tempor` incididunt ut labore et dolore magna aliqua
- [ ] Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi
      ut aliquip ex ea commodo consequat
Normal syntax
```
